backend/internal/api/service_handlers.go

```diff
@@ -222,14 +223,56 @@ func (h *ServiceHandler) getViewportNodes(c *gin.Context) {
 
 // parseVault triggers a vault parsing operation
 func (h *ServiceHandler) parseVault(c *gin.Context) {
-	// TODO: Implement when VaultService is available
-	c.JSON(http.StatusNotImplemented, gin.H{"error": "Vault parsing not yet implemented"})
+	// Use request context
```

Comment:
TODO(CL): Add request body validation for parseVault endpoint. Currently, the endpoint accepts POST requests without any body validation. Consider adding:
1. Optional force flag to force re-parse even if recent parse exists
2. Optional path filter to parse specific directories
3. Request timeout configuration

Example request body:
```json
{
  "force": false,
  "paths": ["/specific/folder"],
  "timeout_seconds": 300
}
```